providers/flagd/pkg/service/service.go

```diff
@@ -50,131 +76,292 @@ type resolutionResponseConstraints interface {
 		schemaV1.ResolveFloatResponse | schemaV1.ResolveObjectResponse
 }
 
-func resolve[req resolutionRequestConstraints, resp resolutionResponseConstraints](
-	ctx context.Context, logger logr.Logger,
-	resolver func(context.Context, *connect.Request[req]) (*connect.Response[resp], error),
-	flagKey string, evalCtx map[string]interface{},
-) (*resp, error) {
-	evalCtxF, err := structpb.NewStruct(evalCtx)
+func (s *Service) Init() error {
+	var err error
+	s.client, err = newClient(s.cfg)
 	if err != nil {
-		logger.Error(err, "struct from evaluation context")
-		return nil, openfeature.NewParseErrorResolutionError(err.Error())
+		return err
 	}
 
-	res, err := resolver(ctx, connect.NewRequest(&req{
-		FlagKey: flagKey,
-		Context: evalCtxF,
-	}))
-	if err != nil {
-		return nil, handleError(err)
-	}
+	ctx, cancelFunc := context.WithCancel(context.Background())
 
-	return res.Msg, nil
+	go func() {
+		s.startEventStream(ctx)
+	}()
+
+	s.cancelHook = cancelFunc
+
+	return nil
+}
```

Comment:
Unless I'm mistaken, we have a similar issue here as we did in Java, where we don't know for sure that the event stream is successfully connected when `Init` is complete.

Can we block here until a ready event?

Reply:
🤦 this is something I always forget about the spec.

I fixed this with the latest commit - https://github.com/open-feature/go-sdk-contrib/pull/317/commits/607ba6e84fcf4b2c278d65a13706ba3eba6518da 

however, this also shows there's some corner cases when it comes to handling events 🤔 something we could improve

Reply:
Yep, that looks good to me.